### PR TITLE
Fix for issue #245, support removal of digits in NDigits2 trigger.

### PR DESCRIPTION
**Pull request by marc1uk**
Loops in WCSimWCTrigger::AlgNDigits and WCSimWCTrigger::FillDigitsCollection now use an iterator to loop over the times map and retrieve the keys present.
In WCSimWCDigi.hh several uses of the operator[] were replaced with map::at.
fDigiComp was removed from the sort in WCSimWCDigi::SortArrayByHitTime as this has no entries at the time when it is called (so switching use to fDigiComp.at() produces an exception). 
#245

### CODE REVIEW

File: include/WCSimWCDigi.hh [hunk 1]
```diff
@@ -139,29 +160,32 @@ public:
   void SortArrayByHitTime() {
```
Comment (tdealtry) [inline include/WCSimWCDigi.hh]: I agree that where `SortArrayByHitTime()` is called currently, we don't need to sort `fDigiComp`. But maybe we want to future proof ourselves for potentially using this sorting method elsewhere in the future by keeping it in. And maybe making the method name more descriptive such as:
```suggestion
  void SortAllMapsByHitTime() {
```
Alternatively, we could do as you suggest and get rid of `fDigiComp` sorting here, and rename to
```suggestion
  void SortMapsByHitTimeForDigitizer() {
```

Comment (tdealtry) [inline include/WCSimWCDigi.hh]: I agree that where `SortArrayByHitTime()` is called currently, we don't need to sort `fDigiComp`. But maybe we want to future proof ourselves for potentially using this sorting method elsewhere in the future by keeping it in. And maybe making the method name more descriptive such as:
```suggestion
  void SortAllMapsByHitTime() {
```
Alternatively, we could do as you suggest and get rid of `fDigiComp` sorting here, and rename to
```suggestion
  void SortMapsByHitTimeForDigitizer() {
```

File: include/WCSimWCDigi.hh [hunk 2]
```diff
@@ -91,21 +91,42 @@ public:
...
-  inline G4float GetPreSmearTime(int gate)   {return time_presmear[gate];};
+  inline G4float GetPe(int gate)     {return pe.at(gate);};
+  inline G4float GetTime(int gate)   {
+    try {
+      return time.at(gate);
+    }
+    catch (...) {
+      G4cerr<<"Exception occurred while attempting to use WCSimWCDigi::GetTime to retrieve time for pe "
+            << gate << " from map of times. The time map has "<<time.size()<<" entries:" << G4endl;
+      for (auto& x: time){
```
Comment (tdealtry) [inline include/WCSimWCDigi.hh]: @eosullivan can comment, but I'm not sure WCSim is using C++11 yet

Comment (tdealtry) [inline include/WCSimWCDigi.hh]: @eosullivan can comment, but I'm not sure WCSim is using C++11 yet

**Comment by tdealtry** [comment]
The Travis build hasn't started for this. Any ideas @brichards64 or @eosullivan?

**Comment by marc1uk** [comment]
Thanks for the feedback. Pushed a new commit to remove c++11 and re-enable sorting of the fDigiComp map, provided it's size is the same as the other maps.

**Comment by tdealtry** [comment]
Great! Thanks for this @marc1uk!

**Pull request merged**